### ISSUE TITLE
Can't get the prompt to display properly (❱❱❱)

### ISSUE THREAD

**Issue by jef** [chauncey-garrett/zsh-prompt-garrett#32]
I edited this part:

```shell script
if (( $SHLVL == 1 )); then
    export PROMPT='${prompt_garrett_altchar_enable}${prompt_garrett_color_prompt}${prompt_garrett_lower_left_corner}${editor_info[keymap]}'
```

and put some `❱❱❱` at the end and it works:

![image](https://user-images.githubusercontent.com/12074633/63316394-b95d3d80-c2dc-11e9-8e6d-ab4f5b4f8ac3.png)

But for whatever reason, the 

```shell script
zstyle ':prezto:module:editor:info:keymap:primary' format "${red}❱%(?.${prompt_garrett_color_prompt}.${red})❱❱ "
```

isn't working on the default.

![image](https://user-images.githubusercontent.com/12074633/63315958-2d96e180-c2db-11e9-8aa0-2c098253e3c0.png)

Could be relevant to sorin-ionescu/prezto#965, but even with those changes, it doesn't work :/

I also cleared all my preferences on iTerm2 and Terminal with a reinstall of zsh just to rule out any funny locale business. Keymap on macOS is English (US).

```shell script
$ locale

LANG="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL=
```

Strange enough, I'm getting my `RPROMPT` though... 

![image](https://user-images.githubusercontent.com/12074633/63316701-caf31500-c2dd-11e9-9ceb-444db0a3a4a6.png)

**Comment by winterhalder** [comment]
I've got the same thing happening - ever since grabbing the latest .zprezto source updates...

**Comment by winterhalder** [comment]
Found a fix - not sure what it's doing - but I noticed these lines had been added into the other prompts...

Add these two lines around ~234 - just after the 3 "add-zsh-hook" calls for pre-exec, precmd and chpwd...

```
  # Tell prezto we can manage this prompt
  zstyle ':prezto:module:prompt' managed 'yes'
```

Then add this line to the bottom of the file...

```
# vim: ft=zsh
```

Honestly, I have no idea what this does - but it worked for me! :)

**Comment by jef** [comment]
> Found a fix - not sure what it's doing - but I noticed these lines had been added into the other prompts...
> 
> Add these two lines around ~234 - just after the 3 "add-zsh-hook" calls for pre-exec, precmd and chpwd...
> 
> ```
>   # Tell prezto we can manage this prompt
>   zstyle ':prezto:module:prompt' managed 'yes'
> ```

That's interesting, because I previously went to the [wiki](https://github.com/sorin-ionescu/prezto/tree/master/modules/prompt#required-variables) and shows the same thing. Maybe we should make a pull request to add those lines in the prompt. But with my experience, I added it but didn't seem to work. _But..._

> 
> Then add this line to the bottom of the file...
> 
> ```
> # vim: ft=zsh
> ```
> 
> Honestly, I have no idea what this does - but it worked for me! :)

That's probably what I was missing... Went to go look up what that means and it seems like it only allows ViM to see what `zsh` files mean (https://github.com/vim/vim/blob/master/runtime/scripts.vim#L83-L85) which makes zero sense why this would work.

I've actually moved on to use _powerlevel10k_, so I don't want to try this, lol. Hopefully our comments help others though!

As a side note, when I was using _garrett_ on the Macbook, it seemed to be completely fine. Same exact setup. But with my Mac Mini it wouldn't work the same.
